### PR TITLE
[TI] initial cc32xx support 

### PR DESCRIPTION
**Pull request by abiradarti**
Co-authored-by: Suyash Jain <s-jain@ti.com>
Co-authored-by: Kobi Leibovitch <kobile@ti.com>

#### Problem
* There was no support for the TI CC32xx platform in this repo

#### Change overview
TI CC32xx platform support 

#### Testing
*lock app functional building and testing

### CODE REVIEW

**Review by woody-apple — APPROVED** [review]
Fast tracking platform changes

File: examples/lock-app/cc32xx/main/debug_settings.h [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+//*****************************************************************************
+// Includes
+//*****************************************************************************
+// Standard includes
+#ifndef DEBUG_SETTINGS_H
```
Comment (andy31415) [inline examples/lock-app/cc32xx/main/debug_settings.h:23]: nit: chip standard is generally using `#pragma once`

consider deleting the `includes` and `Standard includes` comments as they do not seem to help reading the code.

File: src/lwip/cc32xx/arch/cc.h [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+#include <errno.h>
+#include <malloc.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <sys/time.h>
+
+//#include "app_error.h"
```
Comment (andy31415) [inline src/lwip/cc32xx/arch/cc.h:41]: delete commented out code?

File: src/platform/cc32xx/Logging.cpp [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+#include <stdio.h>
+
+using namespace ::chip;
+using namespace ::chip::DeviceLayer;
+using namespace ::chip::DeviceLayer::Internal;
+
+#define DEVICE_LAYER_LOG_BUFFER_SIZE (256)
+static UART2_Handle sDebugUartHandle;
+static char sDebugUartBuffer[DEVICE_LAYER_LOG_BUFFER_SIZE];
+// static pthread_mutex_t mutex;
```
Comment (andy31415) [inline src/platform/cc32xx/Logging.cpp:42]: remove commented out code

**Review by andy31415 — APPROVED** [review]
This is a very large change (5K lines added) so I did not actually do a full review, just a quick glance over.

File: .github/.wordlist.txt [hunk 4]
```diff
@@ -1437,6 +1441,7 @@ xfffff
 xFFFFFFEFFFFFFFFF
 xtensa
 xwayland
+XXSF
```
Comment (andy31415) [inline .github/.wordlist.txt:1444]: XXSF is not a word

File: .github/.wordlist.txt [hunk 5]
```diff
@@ -830,6 +832,7 @@ MLD
 mmevk
 moal
 ModeSelect
+ModelCommand
```
Comment (andy31415) [inline .github/.wordlist.txt:835]: ModelCommand is not a word, others need changes as well (GetWaitDuration is not a word either).

**Pull request merged**